# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.h
# Commit: e488a2686d0b
# Full Hash: e488a2686d0b6354c723dd896c6ce20749c8b3d7
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-12-06 03:46:09
# Regressor Bug: 1766794
# File Overlap Count: 1
# Description:
#   Bug 1802268 part 2: Reinstate partial invalidation of cached HyperText offsets for local HyperTextAccessible. r=eeejay
#   
#   In bug 1766794, I replaced partial invalidation of cached offsets with full cache invalidation.
#   I did this to simplify the code, since the offsets cache was being extended to RemoteAccessible.
#   This is fine for RemoteAccessible because offsets are only needed there when clients query them.
# ==============================================================================

diff -r 80097f40fc50 -r e488a2686d0b accessible/ipc/RemoteAccessibleBase.h
--- a/accessible/ipc/RemoteAccessibleBase.h	Mon Dec 05 23:27:23 2022 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.h	Mon Dec 05 23:27:24 2022 +0000
@@ -369,7 +369,17 @@
   // HyperTextAccessibleBase
   virtual already_AddRefed<AccAttributes> DefaultTextAttributes() override;
 
-  virtual void InvalidateCachedHyperTextOffsets() override {
+  /**
+   * Invalidate cached HyperText offsets. This should be called whenever a
+   * child is added or removed or the text of a text leaf child is changed.
+   * Although GetChildOffset can either fully or partially invalidate the
+   * offsets cache, calculating which offset to invalidate is not worthwhile
+   * because a client might not even query offsets. This is in contrast to
+   * LocalAccessible, where the offsets are always needed to fire text change
+   * events. For RemoteAccessible, it's cheaper overall to just rebuild the
+   * offsets cache when a client next needs it.
+   */
+  void InvalidateCachedHyperTextOffsets() {
     if (mCachedFields) {
       mCachedFields->Remove(nsGkAtoms::offset);
     }
@@ -423,7 +433,7 @@
 
   nsAtom* GetPrimaryAction() const;
 
-  virtual const nsTArray<int32_t>& GetCachedHyperTextOffsets() const override;
+  virtual nsTArray<int32_t>& GetCachedHyperTextOffsets() override;
 
  private:
   uintptr_t mParent;
